internal/middleware/auth.go

```diff
@@ -57,3 +63,39 @@ func AuthenticateRebac(next http.Handler, jimm jujuapi.JIMM) http.Handler {
 		next.ServeHTTP(w, r.WithContext(ctx))
 	}), jimm)
 }
+
+// AuthenticateViaBasicAuth performs basic auth authentication and puts an identity in the request's context
+func AuthenticateViaBasicAuth(next http.Handler, jimm jujuapi.JIMM) http.Handler {
```

Comment:
suggestion: AuthenticateWithSessionTokenViaBasicAuth

naming is hard, but i'd like to convey the message that we're using the session token instead of username/password..